pyogrio/tests/test_raw_io.py

```diff
@@ -647,7 +653,12 @@ def test_write_float_nan_null(tmp_path, dtype):
 
     # set to False
     # by default, GDAL will skip the property for GeoJSON if the value is NaN
-    write(fname, geometry, field_data, fields, **meta, nan_as_null=False)
+    if dtype == "float32":
+        ctx = pytest.warns(RuntimeWarning, match="NaN of Infinity value found. Skipped")
+    else:
+        ctx = contextlib.nullcontext()
```

Comment:
This is bizarre, but for some reason GDAL only raise the warning if the dtype is float32, while the value is actually skipped for both float32 and float64

Reply:
Seems like an issue to raise upstream?

Reply:
Ah, so the issue seems to be that it only warns once, not depending on float32 vs float64. I would expect it to warn once per file, and not once per session, though

https://github.com/OSGeo/gdal/blob/0626cc27486f7ff7ba80e24fe9464e933dbcda2d/ogr/ogrsf_frmts/geojson/ogrgeojsonwriter.cpp#L876-L883